stdlib/public/core/FloatingPoint.swift

```diff
@@ -1890,7 +1890,18 @@ extension BinaryFloatingPoint {
   /// - Parameter value: A floating-point value to be converted.
   @inlinable
   public init<Source: BinaryFloatingPoint>(_ value: Source) {
-    self = Self._convert(from: value).value
+    switch value {
+    case let value_ as Float:
```

Comment:
This should handle `Float16` as well (implemented as `self = Self(Float(value_))` in that case).

Reply:
@stephentyrone The required checks for the availability of `Float16` are...not pretty. See the commit "🤮" for details.